StRoot/StFwdUtils/StFwdAnalysisMaker.cxx

```diff
@@ -49,30 +55,101 @@
+    addHist( new TH1F("ecalEnergy", ";Energy; counts", 100, 0, 15) );
+    addHist( new TH1F("hcalEnergy", ";Energy; counts", 100, 0, 15) );
+    addHist( new TH2F( "ecalXY", ";ecalX;ecalY", 200, -200, 200, 200, -200, 200 ) );
+    addHist( new TH2F( "hcalXY", ";hcalX;hcalY", 200, 0, 50, 200, 0, 50 ) );
+    addHist( new TH1F( "ecaldX", ";dx (trk - ecal); counts", 400, -200, 200 ) );
+    addHist( new TH1F( "matchedEcaldX", ";dx (trk - ecal); counts", 400, -200, 200 ) );
+    addHist( new TH1F( "ecaldY", ";dy (trk - ecal); counts", 400, -200, 200 ) );
+    addHist( new TH1F( "matchedEcaldY", ";dy (trk - ecal); counts", 400, -200, 200 ) );
+    addHist( new TH1F( "ecaldR", ";dr (trk - ecal); counts", 400, 0, 400 ) );
+    addHist( new TH1F( "ecalMindR", ";dr (trk - ecal); counts", 400, 0, 400 ) );
+    addHist( new TH1F( "matchedEcaldR", ";dr (trk - ecal); counts", 400, 0, 400 ) );
+    addHist( new TH1F( "hcaldX", ";dx (trk - hcal); counts", 400, -200, 200 ) );
+    addHist( new TH2F( "hcaldXdNFit", ";dx (trk - hcal); nFit", 400, -200, 200, 10, 0, 10 ) );
+    addHist( new TH1F( "matchedHcaldX", ";dx (trk - hcal); counts", 400, -200, 200 ) );
+    addHist( new TH1F( "hcaldY", ";dy (trk - hcal); counts", 400, -200, 200 ) );
+    addHist( new TH2F( "hcaldYdNFit", ";dy (trk - hcal); nFit", 400, -200, 200, 10, 0, 10 ) );
+    addHist( new TH1F( "matchedHcaldY", ";dy (trk - hcal); counts", 400, -200, 200 ) );
+    addHist( new TH1F( "hcaldR", ";dr (trk - hcal); counts", 400, 0, 400 ) );
+    addHist( new TH1F( "hcalMindR", ";dr (trk - hcal); counts", 400, 0, 400 ) );
+    addHist( new TH1F( "matchedHcaldR", ";dr (trk - hcal); counts", 400, 0, 400 ) );
+    addHist( new TH2F( "trkEcalX", ";trkX;ecalX", 300, -150, 150, 300, -150, 150 ) );
+    addHist( new TH2F( "trkEcalY", ";trkY;ecalY", 300, -150, 150, 300, -150, 150 ) );
+    addHist( new TH2F( "trkEcalMinX", ";trkX;ecalX", 300, -150, 150, 300, -150, 150 ) );
+    addHist( new TH2F( "trkEcalMinY", ";trkY;ecalY", 300, -150, 150, 300, -150, 150 ) );
+    addHist( new TH2F( "trkHcalX", ";trkX;hcalX", 300, -150, 150, 300, -150, 150 ) );
+    addHist( new TH2F( "trkHcalY", ";trkY;hcalY", 300, -150, 150, 300, -150, 150 ) );
+    addHist( new TH2F( "trkHcalMinX", ";trkX;hcalX", 300, -150, 150, 300, -150, 150 ) );
+    addHist( new TH2F( "trkHcalMinY", ";trkY;hcalY", 300, -150, 150, 300, -150, 150 ) );
+
+    return kStOK;
+}
 //________________________________________________________________________
 int StFwdAnalysisMaker::Make() {
     LOG_DEBUG << "StFwdAnalysisMaker::Make" << endm;
     StEvent *event = (StEvent *)GetDataSet("StEvent");
-    if (!event){
-        LOG_INFO << "Cannot find StEvent" << endm;
-        return kStOK;
+    if (event){
+        StFttCollection *fttCol = event->fttCollection();
+        if (fttCol){
+            LOG_INFO << "The Ftt Collection has " << fttCol->numberOfPoints() << " points" << endm;
+        }
```

Comment:
Maybe better to move this into the `ProcessFwdTracks()` routine and have a similar report from the MuDst in `ProcessFwdMuTracks()`? As it is, you obtain the StEvent pointer twice when using StEvent, and cast it differently the two times you obtain it. Not critical, but odd.

Reply:
moved in [030a2f5](https://github.com/star-bnl/star-sw/pull/636/commits/030a2f51a0cb1aa2368fac8ec6070243646aeda0)